server/venueless/api/task.py

```diff
@@ -0,0 +1,139 @@
+            result.append(st.short_token)
+            bulk_create.append(st)
+
+    if not long:
+        ShortToken.objects.bulk_create(bulk_create)
+    return result
+
+
+def generate_talk_token(video_settings, video_tokens, event_slug):
+    """
+    Generate talk token
+    :param video_settings: A dictionary representing the video settings
+    :param video_tokens: A list of video tokens
+    :param event_slug:  A string representing the event slug
+    :return: A token
+    """
+    iat = dt.datetime.utcnow()
+    exp = iat + dt.timedelta(days=30)
+    payload = {
+        "exp": exp,
+        "iat": iat,
+        "video_tokens": video_tokens,
+        "slug": event_slug,
+    }
+    token = jwt.encode(payload, video_settings.get("secret"), algorithm="HS256")
+    return token
+
+
+@shared_task(bind=True, max_retries=5, default_retry_delay=60)
+def configure_video_settings_for_talks(
+    self, world_id, days, number, traits, long=False
+):
+    """
+    Configure video settings for talks
+    :param self: instance of the task
+    :param world_id: A integer representing the world id
+    :param days: A integer representing the number of days the token is valid
+    :param number: A integer representing the number of tokens to generate
+    :param traits: A dictionary representing the traits of the token
+    :param long: A boolean representing if the token is long or short
+    """
+    try:
+        if not isinstance(world_id, str) or not world_id.isalnum():
+            raise ValueError("Invalid world_id format")
+        world = World.objects.get(id=world_id)
+        event_slug = world_id
+        jwt_secrets = world.config.get("JWT_secrets", [])
+        if not jwt_secrets:
+            logger.error("JWT_secrets is missing or empty in the configuration")
+            return
+        jwt_config = jwt_secrets[0]
+        video_tokens = generate_video_token(world, days, number, traits, long)
+        talk_token = generate_talk_token(jwt_config, video_tokens, event_slug)
+        header = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {talk_token}",
+        }
+        payload = {
+            "video_settings": {
+                "audience": jwt_config.get("audience"),
+                "issuer": jwt_config.get("issuer"),
+                "secret": jwt_config.get("secret"),
+            }
+        }
+        requests.post(
+            "{}/api/configure-video-settings/".format(settings.EVENTYAY_TALK_BASE_PATH),
+            json=payload,
+            headers=header,
+        )
+        world.config["pretalx"] = {
+            "event": event_slug,
+            "domain": "{}".format(settings.EVENTYAY_TALK_BASE_PATH),
+            "pushed": datetime.datetime.now().isoformat(),
+            "connected": True,
+        }
+        world.save()
+    except requests.exceptions.ConnectionError as e:
+        logger.error("Connection error: %s", str(e))
```

Comment:
You don't need to wrap `e` with `str`. When the log message is to be built, the `%` formatter will tell `logging` to "stringify" the passed arguments.
Wrapping `str` here is even undesired, it makes "stringify" operation run before we need to build the log message.